### PR TITLE
schema: Validate GH & BBS config have repos OR repositoryQuery

### LINKED ISSUE #4322: Why is repositoryQuery required when orgs is set?
image

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes #4322 by changing the validation of external service
configurations for GitHub and Bitbucket Server to validate that either
repositoryQuery OR repos are set.

Questions for @tsenart & @keegancsmith:
1. Is this the right place to put the validators?
2. Do I also need to remove `GithubSetDefaultRepositoryQueryMigration` and `BitbucketServerSetDefaultRepositoryQueryMigration`?

### CODE REVIEW

File: cmd/frontend/db/external_services.go [hunk 1]
```diff
@@ -32,6 +32,33 @@ type ExternalServicesStore struct {
 	BitbucketServerValidators []func(*schema.BitbucketServerConnection, []schema.AuthProviders) error
 }
 
+// NewExternalServicesStore returns an ExternalServicesStore with
```
Comment (tsenart) [inline cmd/frontend/db/external_services.go]: I think we can accomplish this validation with JSON schema alone.

Comment (tsenart) [reply cmd/frontend/db/external_services.go]: `"anyOf": [{ "required": ["repos"] }, { "required": ["repositoryQuery"] }],

Comment (mrnugget) [reply cmd/frontend/db/external_services.go]: Perfect! I was wondering about that this morning. Thank you.

**Comment by mrnugget** [comment]
@keegancsmith @tsenart Removed the two migrations relating to `repositoryQuery`. I'd do the "remove all migrations" in a different PR?

**Pull request merged**